oodeel/extractor/keras_feature_extractor.py

```diff
@@ -190,9 +201,28 @@ def predict(
                     f if features[i] is None else tf.concat([features[i], f], axis=0)
                 )
 
+            # Get labels
+            if return_labels and contains_labels:
+                lbl_batch = elem[1]  # labels must be at index 1 in the batch tuple
+                # If labels are one-hot encoded, take the argmax
+                if tf.rank(lbl_batch) > 1 and lbl_batch.shape[1] > 1:
+                    lbl_batch = tf.reshape(lbl_batch, shape=[lbl_batch.shape[0], -1])
+                    lbl_batch = tf.argmax(lbl_batch, axis=1)
+                # If labels are in two dimensions, squeeze them
+                if len(lbl_batch.shape) > 1:
+                    lbl_batch = tf.reshape(lbl_batch, [lbl_batch.shape[0]])
```

Comment:
Maybe make a function out of it and use it twice, here and in the case of  `isinstance(dataset, get_args(ItemType)) == True`